### PR TITLE
parameterize and better organize shellspec tests

### PR DESCRIPTION
**Pull request by jacobsalmela**
# Summary and Scope

- better parameterizes shellspec tests utilizing vars and shellspec `Parameters:`
- organizes shellpsec tests a bit better
- uses `/tmp/.cani/` for all tests instead of a relative path (helps for consistency and to avoid clash with developer builds)
- changes `make spec` to use the same script as integration and edge.  this required moving the existing tests into a new `spec/functional` directory.  This also moves the script into `spec/support/bin`, which is a better home for it
- updates the custom hardware fixture

# Risks and Mitigations
 

Low

### CODE REVIEW

File: spec/functional/cani_add_blade_spec.sh [hunk 1]
```diff
@@ -210,37 +220,42 @@ Describe 'cani add blade'
...
+  # The line 4 of stderr should include "Suggested NodeBlade number: $5"
   The line 6 of stderr should include 'NodeBlade was successfully staged to be added to the system'
-  The line 7 of stderr should include 'UUID: '
-  The line 8 of stderr should include "Cabinet: $3"
-  The line 9 of stderr should include "Chassis: $4"
-  The line 10 of stderr should include "Blade: $5"
+  # The line 7 of stderr should include 'UUID: '
+  # The line 8 of stderr should include "Cabinet: $3"
+  # The line 9 of stderr should include "Chassis: $4"
+  # The line 10 of stderr should include "Blade: $5"
```
Comment (shunr-hpe) [inline spec/functional/cani_add_blade_spec.sh:258]: Possibly removed the commented out lines. However, if they are useful or informative in anyway they should probably be kept.

File: spec/functional/cani_session_spec.sh [hunk 2]
```diff
@@ -24,65 +24,60 @@
...
 #  - no connection to SLS
-It '(no connection to provider) --config canitest.yml init csm'
+It "(no connection to provider) --config $CANI_CONF init csm"
   BeforeCall remove_config
   BeforeCall remove_datastore
-  When call bin/cani alpha session --config canitest.yml init csm
+  When call bin/cani alpha session --config "$CANI_CONF" init csm
   The status should equal 1
-  The line 1 of stderr should include 'canidb.json does not exist, creating default datastore'
+  The line 1 of stderr should include '/tmp/.cani/canidb.json does not exist, creating default datastore'
```
Comment (shunr-hpe) [inline spec/functional/cani_session_spec.sh:80]: If $CANI_CONF can be configured then make the check
'.cani/canidb.json does not exist, creating default datastore'

File: spec/spec_helper.sh [hunk 3]
```diff
@@ -34,9 +34,16 @@ spec_helper_precheck() {
...
   setenv FIXTURES="$SHELLSPEC_HELPERDIR/testdata/fixtures"
-  
+  # use /tmp for consistent test dir that doesn't conflict with local dev
+  # also helpful for cani config fixtures if this is a static, abs path
+  setenv CANI_DIR="/tmp/.cani"
+  setenv CANI_CONF="/tmp/.cani/cani.yml"
+  setenv CANI_DS="/tmp/.cani/canidb.json"
+  setenv CANI_LOG="/tmp/.cani/canidb.log"
+  setenv CANI_CUSTOM_HW_DIR="/tmp/.cani/hardware-types"
+  setenv CANI_CUSTOM_HW_CONF="/tmp/.cani/hardware-types/my_custom_hw.yml"
```
Comment (shunr-hpe) [inline spec/spec_helper.sh:46]: Can these make use of the CANI_DIR env?
`="$CANI_DIR/`

**Review by shunr-hpe — APPROVED** [review]

File: spec/functional/cani_session_spec.sh (hunk 2 above)
Comment (jacobsalmela) [reply spec/functional/cani_session_spec.sh]: fixed

**Review by shunr-hpe — APPROVED** [review]

**Review by rsjostrand-hpe — APPROVED** [review]

**Pull request merged**